src/styles/main.scss

```diff
@@ -1,3 +1,57 @@
-body {
-  margin: 0;
+@import "./utils/variables.scss";
+@import "./utils/mixins.scss";
+@import "./reset.scss";
+
+@for $i from 1 to 32 {
+  .calendar__day:nth-child(#{$i})::before {
+    content: "#{$i}";
+  }
+}
+
+$days:
+  "mon" 0,
+  "tue" 103px,
+  "wed" 206px,
+  "thu" 309px,
+  "fri" 412px,
+  "sat" 515px,
+  "sun" 618px;
+
+@each $name, $margin-size in $days {
+  .calendar__start-day--#{$name} >:first-child {
+    margin-left: #{$margin-size};
+  }
+}
+
+@for $i from 28 to 31 {
+  .calendar__month-length--#{$i} >:nth-child(n + #{$i + 1}) {
+    display: none;
+  }
+}
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  max-width: calc( (#{$block-size} + 1px) * 7 + (#{$calendar-padding} * 2) );
+  gap: 1px;
+  padding: $calendar-padding;
+  margin: auto;
+
+  &__day {
+    @include square($block-size);
+    background-color: $block-color;
+    border: 1px solid black;
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    font-style: arial, sans-serif;
+    font-size: 30px;
+
+    &:hover {
+      background-color: pink;
+      cursor: pointer;
+      transform: translateY(-20px);
+      transition-duration: 0.5s;
```

Comment:
`cursor` and `transition-duration` should be in `&__day`. add `transition-property` to specify what do you want to animate